### PR TITLE
Update rubygem-radcli to support SCL

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: packages/plugins/rubygem-radcli/rubygem-radcli.spec [hunk 1]
```diff
@@ -1,87 +1,104 @@
...
 License: Artistic-2.0
-URL: https://github.com/martencassel/radcli
-Source0: https://rubygems.org/downloads/%{gem_name}-%{version}.gem
-Requires: ruby(release)
-Requires: ruby
-Requires: ruby(rubygems)
-BuildRequires: ruby(release)
-BuildRequires: ruby-devel
-BuildRequires: rubygems-devel
+URL: http://github.com/martencassel/radcli
```
Comment (evgeni) [inline packages/plugins/rubygem-radcli/rubygem-radcli.spec]:
```suggestion
URL: https://github.com/martencassel/radcli
```

File: packages/plugins/rubygem-radcli/rubygem-radcli.spec [hunk 2]
```diff
@@ -1,87 +1,104 @@
...
+Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem
+
+# start specfile generated dependencies
+Requires: %{?scl_prefix_ruby}ruby(release)
+Requires: %{?scl_prefix_ruby}ruby
+Requires: %{?scl_prefix_ruby}ruby(rubygems)
+Requires: %{?scl_prefix}rubygem(rake-compiler)
```
Comment (evgeni) [inline packages/plugins/rubygem-radcli/rubygem-radcli.spec]:
```suggestion
BuildRequires: %{?scl_prefix}rubygem(rake-compiler)
```

File: packages/plugins/rubygem-radcli/rubygem-radcli.spec [hunk 3]
```diff
@@ -1,87 +1,104 @@
...
 %install
 mkdir -p %{buildroot}%{gem_dir}
-cp -a .%{gem_dir}/* \
+cp -pa .%{gem_dir}/* \
         %{buildroot}%{gem_dir}/
 
-mkdir -p %{buildroot}%{gem_extdir_mri}/lib
-cp -a %{buildroot}%{gem_instdir}/lib/radcli.so %{buildroot}%{gem_extdir_mri}/lib/
-
-# rake-compiler isn't needed on the system itself
```
Comment (evgeni) [inline packages/plugins/rubygem-radcli/rubygem-radcli.spec]: I think this is correct and should remain.

File: packages/plugins/rubygem-radcli/rubygem-radcli.spec [hunk 4]
```diff
@@ -1,87 +1,104 @@
...
-sed -i '/rake-compiler/ s/runtime/development/' %{buildroot}/%{gem_spec}
+mkdir -p %{buildroot}%{gem_extdir_mri}
+cp -a .%{gem_extdir_mri}/{gem.build_complete,*.so} %{buildroot}%{gem_extdir_mri}/
```
Comment (evgeni) [inline packages/plugins/rubygem-radcli/rubygem-radcli.spec]: this seems not to work

**Comment by ehelms** [comment]
@evgeni I am quite confused about the rake-compiler. The rubygems.org lists it as a runtime dependency. Why wouldn't I want it then as a runtime dependency?

File: packages/plugins/rubygem-radcli/rubygem-radcli.spec [hunk 5]
```diff
@@ -1,87 +1,103 @@
...
-# rake-compiler isn't needed on the system itself
 sed -i '/rake-compiler/ s/runtime/development/' %{buildroot}/%{gem_spec}
 
-# Prevent dangling symlink in -debuginfo (rhbz#878863).
-rm -rf %{buildroot}%{gem_instdir}/ext/
-
 %files
 %dir %{gem_instdir}
 %{gem_extdir_mri}
 %{gem_instdir}/CHANGES
```
Comment (ekohl) [inline packages/plugins/rubygem-radcli/rubygem-radcli.spec]:
```suggestion
%doc %{gem_instdir}/CHANGES
```

File: packages/plugins/rubygem-radcli/rubygem-radcli.spec (hunk 3 above)
Comment (ekohl) [reply packages/plugins/rubygem-radcli/rubygem-radcli.spec]: Could you keep the comment?

File: packages/plugins/rubygem-radcli/rubygem-radcli.spec [hunk 6]
```diff
@@ -1,87 +1,103 @@
...
 # %%gem_install compiles any C extensions and installs the gem into ./%%gem_dir
 # by default, so that we can move it into the buildroot in %%install
+%{?scl:scl enable %{scl} - << \EOF}
 %gem_install
+%{?scl:EOF}
 
 %install
 mkdir -p %{buildroot}%{gem_dir}
-cp -a .%{gem_dir}/* \
+cp -pa .%{gem_dir}/* \
```
Comment (ekohl) [inline packages/plugins/rubygem-radcli/rubygem-radcli.spec]: We've simplified the tempate in f02a98ade2c230f3968cd12e0ac220262f10109c:
```suggestion
cp -a .%{gem_dir}/* \
```

File: packages/plugins/rubygem-radcli/rubygem-radcli.spec [hunk 7]
```diff
@@ -1,87 +1,103 @@
+# Generated from radcli-1.0.0.gem by gem2rpm -*- rpm-spec -*-
```
Comment (ekohl) [inline packages/plugins/rubygem-radcli/rubygem-radcli.spec]: We've dropped this line in the template:
```suggestion
```

File: packages/plugins/rubygem-radcli/rubygem-radcli.spec [hunk 8]
```diff
@@ -1,87 +1,103 @@
...
 %files doc
 %doc %{gem_docdir}
 %doc %{gem_instdir}/README.md
 %{gem_instdir}/Rakefile
+%{gem_instdir}/radcli.gemspec
 %{gem_instdir}/test
 
 %changelog
+* Tue Sep 17 2019 Eric D. Helms <ericdhelms@gmail.com> 1.0.0-2
+- Update to 1.0.0-2
```
Comment (ekohl) [inline packages/plugins/rubygem-radcli/rubygem-radcli.spec]: Might be nice to write something about the SCL template here

**Comment by evgeni** [comment]
@ehelms because you don't compile the gem at runtime ;-)

Ruby Gems (contrary to e.g. Python modules) don't have the concept of "install dependency", but this is really what "rake-compiler" is, it extends default rake tasks for building native gems and as thus only required during the install phase. You will see that there is no `require 'rake-compiler'` anywhere in the code (and that gem does not even have a `lib` folder, only `tasks`).

**Comment by ehelms** [comment]
Updated to handle both nonscl and SCL

File: packages/plugins/rubygem-radcli/rubygem-radcli.spec [hunk 9]
```diff
@@ -64,24 +83,29 @@ rm -rf %{buildroot}%{gem_instdir}/ext/
...
 %{gem_instdir}/build_adcli.sh
-%exclude %{gem_instdir}/radcli.gemspec
 %{gem_instdir}/scripts
-%{gem_instdir}/lib
 %exclude %{gem_cache}
 %{gem_spec}
+%exclude %{gem_instdir}/radcli.gemspec
+%if 0%{!?scl:1}
+%{gem_extdir_mri}
+%{gem_instdir}/lib
```
Comment (ekohl) [inline packages/plugins/rubygem-radcli/rubygem-radcli.spec]: This is equivalent to `%{gem_libdir}` in `/etc/rpm/macros.rubygems.rh-ruby25`

```suggestion
%{gem_libdir}
```

**Pull request merged**